### PR TITLE
Fix duplicate leave events

### PR DESCRIPTION
**Pull request by S7evinK**
Still investigating, but doesn't even seem to require multiple pollers as per https://github.com/matrix-org/sliding-sync/issues/250

### CODE REVIEW

File: tests-e2e/membership_transitions_test.go [hunk 1]
```diff
@@ -397,6 +398,140 @@ func TestInviteRejectionTwice(t *testing.T) {
...
+			bob.JoinRoom(t, inviteRoomID, []string{})
+			res = bob.SlidingSyncUntilMembership(t, res.Pos, inviteRoomID, bob, "join")
+			t.Logf("bob joined")
+
+			// leave the room again, we should receive exactly one leave response
+			bob.LeaveRoom(t, inviteRoomID)
+			res = bob.SlidingSyncUntilMembership(t, res.Pos, inviteRoomID, bob, "leave")
+
+			if room, ok := res.Rooms[inviteRoomID]; ok {
+				// If alice is NOT syncing, we run into this failure mode
```
Comment (kegsay) [inline tests-e2e/membership_transitions_test.go:444]: Can you confirm what part of this test was failing without the diff to `user.go`?

File: tests-e2e/membership_transitions_test.go [hunk 2]
```diff
@@ -397,6 +398,140 @@ func TestInviteRejectionTwice(t *testing.T) {
...
+					t.Errorf("[multiple poller=%v] expected 1 timeline event, got %d", aliceSyncing, c)
+				}
+			} else {
+				t.Errorf("[multiple poller=%v] expected room %s in response, but didn't find it", aliceSyncing, inviteRoomID)
+			}
+
+			res = bob.SlidingSync(t, sync3.Request{}, WithPos(res.Pos))
+
+			// this should not happen, as we already send down the leave event
+			// If alice is synching, we run into this failure mode
```
Comment (kegsay) [inline tests-e2e/membership_transitions_test.go:458]:
```suggestion
			// If alice is syncing, we run into this failure mode
```

File: tests-e2e/membership_transitions_test.go [hunk 3]
```diff
@@ -397,6 +398,140 @@ func TestInviteRejectionTwice(t *testing.T) {
...
```
Comment (kegsay) [inline tests-e2e/membership_transitions_test.go:524]:
```suggestion
			// If alice is syncing, we run into this failure mode
```

**Review by kegsay — APPROVED** [review]

File: tests-e2e/membership_transitions_test.go (hunk 1 above)
Comment (S7evinK) [reply tests-e2e/membership_transitions_test.go:444]: If Alice is _NOT_ syncing, we receive the leave event twice, once through  `Accumulate` and once through `OnLeftRoom`. 

If Alice _IS_ syncing however, the test fails for exactly the same reason - we receive the same event twice, but at a different place in the test. We're syncing as Bob until we left the room, after that we do another `SlidingSync` call, which again receives the event.

**Pull request merged**